### PR TITLE
Separated http methods

### PR DESCRIPTION
**Pull request by cumason123**
Addresses #59 

## Changes
Separated http methods from CloudEvent class. Also instantiated http module directory

## One line description for the changelog
Separated http methods from CloudEvent class and instantiated http module directory

Feel free to make suggestions for how users should preferably interact with out http methods. I wrote this to more or less open up conversation. Look at the README CloudEvent usage for a better understanding on the changes to the http methods.

- [*] Tests pass
- [*] Appropriate changes to README are included in PR

### CODE REVIEW

File: samples/http-cloudevents/client.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def send_binary_cloud_event(url):
     headers, body = to_binary_http(event)
 
     # send and print event
-    requests.post(url, headers=headers, data=body)
+    requests.post(url, data=body, headers=headers)
```
Comment (grant) [inline samples/http-cloudevents/client.py:34]: I'd probably keep headers in front of the body as is natural in a HTTP request.

File: samples/http-cloudevents/server.py [hunk 2]
```diff
@@ -23,6 +23,18 @@
 def home():
     # create a CloudEvent
     event = from_http(request.get_data(), request.headers)
+    print(event)
```
Comment (grant) [inline samples/http-cloudevents/server.py:26]: What does the output look like?

If we print the whole event, do we really need to print `event['source']`, `event['id']` etc?

File: samples/http-cloudevents/server.py [hunk 3]
```diff
@@ -23,6 +23,18 @@
...
+    # you can access cloudevent fields as seen below
+    print(
+        f"Found CloudEvent from {event['source']} with specversion {event['specversion']}"
+    )
+
+    if event["type"] == "com.example.sampletype1":
+        print(f"CloudEvent {event['id']} is binary")
+
+    elif event["type"] == "com.example.sampletype2":
+        print(f"CloudEvent {event['id']} is structured")
```
Comment (grant) [inline samples/http-cloudevents/server.py]: Not sure if the event type should determine if it's binary or structured.

I'd probably just remove this as we already print the whole event.

Comment (cumason123) [reply samples/http-cloudevents/server.py]: Truth be told, I only added that if statement as a means of showing the user how to check the event's type and id given we don't have getters/setters on that in CloudEvent yet. I would still like to show case member access on those required fields so suggestions on that is welcome.

File: samples/http-cloudevents/server.py (hunk 2 above)
Comment (cumason123) [reply samples/http-cloudevents/server.py:26]: I was mostly trying to showcase access to event attributes. If anything, I'd remove the print(event). Open to thoughts on this!

**Pull request merged**